### DISCUSSION TITLE
Large difference in backed up resources with Scheduled vs Manual backup

### DISCUSSION THREAD

**Discussion by ghost** [vmware-tanzu/velero#3938] [Community support Q&A]
Hello!

**Update:** Velero automated backups working as intended. It was an operator issue. 🤦 Sorry for the noise here!

I setup Velero using IAM Roles for Service Accounts following this blog: https://www.fourco.nl/blogs/backup-and-restore-a-kubernetes-cluster-with-velero/

However, I am seeing a very large difference between the number of backed up resources from a **Scheduled** velero backup vs when I **manually** execute `velero backup create ...`. Both have the option to select all resources on the given cluster.

For an example **scheduled** backup, I am seeing `Items backed up: 4251`, however if I execute the **manual** backup with the same configuration, I see `Items backed up: 5175`.

Furthermore, and the bigger issue, is when I extract the `.tar.gz` file generated by the Velero **manual** backup, I see a list of all resources in the backup. When I extract the `.tar.gz` file in the **scheduled** backup, it is only a `version` file with no resources.

I'd like to think this has something to do with misconfiguration of permissions, but the scheduled operation reports no errors and successful completion.

I am running the following version of Velero:
```
$ velero version
Client:
	Version: v1.6.1
	Git commit: -
Server:
	Version: v1.6.1
```

I am using `velero-plugin-for-aws:v1.2.0`. Velero is running on an EKS v1.18 cluster and pushing backups to S3 without issue.

***Scheduled backup describe*:**
```
$ velero backup describe velero-daily-backups-20210708000043
Name:         velero-daily-backups-20210708000043
Namespace:    velero
Labels:
              helm.sh/chart=velero-2.23.0
              velero.io/schedule-name=velero-daily-backups
              velero.io/storage-location=default
Annotations:
              velero.io/source-cluster-k8s-gitversion=v1.18.9-eks-d1db3c
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=18+
Phase:  Completed
Errors:    0
Warnings:  0
Namespaces:
  Included:  *
  Excluded:  <none>
Resources:
  Included:        *
  Excluded:        <none>
  Cluster-scoped:  auto
Label selector:  <none>
Storage Location:  default
Velero-Native Snapshot PVs:  auto
TTL:  360h0m0s
Hooks:  <none>
Backup Format Version:  1.1.0
Started:    2021-07-07 20:00:43 -0400 EDT
Completed:  2021-07-07 20:00:53 -0400 EDT
Expiration:  2021-07-22 20:00:43 -0400 EDT
Total items to be backed up:  4251
Items backed up:              4251
Velero-Native Snapshots: <none included>
```

***Manual backup describe*:**
```
$ velero backup describe steven-testing-production-2-1-1
Name:         steven-testing-production-2-1-1
Namespace:    velero
Labels:       velero.io/storage-location=default
Annotations:  velero.io/source-cluster-k8s-gitversion=v1.18.9-eks-d1db3c
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=18+
Phase:  Completed
Errors:    0
Warnings:  0
Namespaces:
  Included:  *
  Excluded:  <none>
Resources:
  Included:        *
  Excluded:        <none>
  Cluster-scoped:  auto
Label selector:  <none>
Storage Location:  default
Velero-Native Snapshot PVs:  auto
TTL:  720h0m0s
Hooks:  <none>
Backup Format Version:  1.1.0
Started:    2021-07-08 13:42:15 -0400 EDT
Completed:  2021-07-08 13:42:25 -0400 EDT
Expiration:  2021-08-07 13:42:15 -0400 EDT
Total items to be backed up:  5175
Items backed up:              5175
Velero-Native Snapshots: <none included>
```

Since there are no resources in the **scheduled** backups, they are useless at this point. Any guidance would be highly appreciated.

Thank you!

**Comment by zubron** [comment]
It would interesting to know _which_ resources are being missed in the scheduled backups. Is there anything that stands out if you compare the two backups using the `--details` flag (`velero backup describe --details`)? Also, is there any chance that new resources have been introduced in the time period between the daily scheduled backup and the manual one taken given that they are almost 18 hours apart?

**Reply by ghost** [reply]
I was looking at the incorrect backup snapshot, one that had failed so it was indeed empty. The automated backup does work as intended. I deeply apologize for the noise! Thank you for the `--details` hint as well!
